cli/azd/pkg/project/framework_service_external.go

```diff
@@ -0,0 +1,531 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package project
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"sync"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/async"
+	"github.com/azure/azure-dev/cli/azd/pkg/azdext"
+	"github.com/azure/azure-dev/cli/azd/pkg/extensions"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/google/uuid"
+)
+
+// externalTool implements the tools.ExternalTool interface for extension-provided tools
+type externalTool struct {
+	name       string
+	installUrl string
+}
+
+func (et *externalTool) CheckInstalled(ctx context.Context) error {
+	// Extension-provided tools are assumed to be checked by the extension itself
+	return nil
+}
+
+func (et *externalTool) InstallUrl() string {
+	return et.installUrl
+}
+
+func (et *externalTool) Name() string {
+	return et.name
+}
```

Comment:
Lets think about how we can support external tool checks - right now I don't think this is fully wired up.